### PR TITLE
Update manylinux build scripts

### PR DESCRIPTION
**Pull request by snnn**
**Description**: 

Update manylinux build scripts. Sync it with the latest upstream.

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: tools/ci_build/github/linux/docker/build_scripts/README.md [hunk 1]
```diff
@@ -7,13 +7,22 @@ diff -r /data/bt/os/manylinux/docker/build_scripts/install-entrypoint.sh ./insta
 > yum versionlock cuda* libcudnn*
 \ No newline at end of file
 diff -r /data/bt/os/manylinux/docker/build_scripts/install-runtime-packages.sh ./install-runtime-packages.sh
-87c87,92
-<       TOOLCHAIN_DEPS="devtoolset-9-binutils devtoolset-9-gcc devtoolset-9-gcc-c++ devtoolset-9-gcc-gfortran"
+87c87,95
```
Comment (tianleiwu) [inline tools/ci_build/github/linux/docker/build_scripts/README.md:10]: Need resolve merge here?

Comment (snnn) [reply tools/ci_build/github/linux/docker/build_scripts/README.md:10]: It's a readme file. To note what I changed.

File: tools/ci_build/github/linux/docker/build_scripts/finalize.sh [hunk 2]
```diff
@@ -20,16 +20,20 @@ for PREFIX in $(find /opt/_internal/ -mindepth 1 -maxdepth 1 -name 'cpython*');
 	if [ -e ${PREFIX}/bin/pip3 ] && [ ! -e ${PREFIX}/bin/pip ]; then
 		ln -s pip3 ${PREFIX}/bin/pip
 	fi
+	PY_VER=$(${PREFIX}/bin/python -c "import sys; print('.'.join(str(v) for v in sys.version_info[:2]))")
```
Comment (jcwchen) [inline tools/ci_build/github/linux/docker/build_scripts/finalize.sh:23]: nit: Is `PY_VER` same as `PYVERS` at line 31? (It came from their original code...)

**Review by jcwchen — APPROVED** [review]
Overall LGTM. Thanks for the update

File: tools/ci_build/github/linux/docker/build_scripts/finalize.sh (hunk 2 above)
Comment (snnn) [reply tools/ci_build/github/linux/docker/build_scripts/finalize.sh:23]: I think it should be. But that's their code. :-( 
These two lines are slightly different.

**Pull request merged**